frontend/src/stores/useProjectStore.ts

```diff
@@ -57,6 +75,10 @@ export const createNewProject = (projectType: ProjectType = DEFAULT_PROJECT_TYPE
   }
 })
 
+const nextIDFor = (elementArr: AutomataState[] | BaseAutomataTransition[] | ProjectComment[]): number => {
+  return 1 + Math.max(-1, ...elementArr.map((e: AutomataState | BaseAutomataTransition | ProjectComment) => e.id))
+}
```

Comment:
We only need an `id` so we can just specify that (Also means it will work for any future items)
```suggestion
const nextIDFor = (elementArr: {id?: number}[]): number => {
  // We can't do elementArr.length + 1 since that might reuse an ID from a deleted item
  // Uses || -1 to handle any of the items being `undefined` (Shouldn't happen, but our type says it can)
  return 1 + (Math.max(-1, ...elementArr.map(e => e.id)) || -1)
}
```
I used `|| -1` since `Math.max(-1, undefined)` returns `NaN` (This shouldn't happen in practise though, except the type for `ProjectComment` needs to allow for undefined ID so this is more just defensive programming).

Also I can do this if you want, but `createState`, `createTransition`, and `createComment` should also be using `nextIDFor`

Reply:
I tried putting this function in, but it resulted in some unexpected behaviour: if I have an empty project, create a state (q0), and then copy paste it the next ID is not calculated correctly (it is pasted as q0 as well). 

So I was thinking, perhaps it could be left as is and this could be opened as a separate issue in the interest of prioritising getting the templates into dev? I'm sure it wouldn't be too hard to work out what the issue is and get this refactored version into the code, I am just keen to get this merged if possible. Perhaps calling the function in createState, createTransition, and createTransition could be tackled in that same issue? Let me know what you think